### PR TITLE
[IOTDB-1565] [To rel/0.12] Add sql: set system to readonly/writable

### PR DESCRIPTION
**Pull request by HTHou**
## Description

Manually set IoTDB system to read-only or writable status.

```sql
IoTDB> SET SYSTEM TO READONLY
IoTDB> SET SYSTEM TO WRITABLE
```

Here is an example.
```sql
IoTDB> insert into root.sg.d1(time,s1) values(100,2)
Msg: The statement is executed successfully.
IoTDB> set system to readonly
Msg: The statement is executed successfully.
IoTDB> insert into root.sg.d1(time,s1) values(200,2)
Msg: 411: Current system mode is read-only, does not support non-query operation
IoTDB> flush
Msg: The statement is executed successfully.
IoTDB> select * from root
+-----------------------------+-------------+
|                         Time|root.sg.d1.s1|
+-----------------------------+-------------+
|1970-01-01T08:00:00.100+08:00|          2.0|
+-----------------------------+-------------+
Total line number = 1
It costs 0.273s
IoTDB> set system to writable
Msg: The statement is executed successfully.
IoTDB> insert into root.sg.d1(time,s1) values(200,2)
Msg: The statement is executed successfully.
IoTDB> select * from root
+-----------------------------+-------------+
|                         Time|root.sg.d1.s1|
+-----------------------------+-------------+
|1970-01-01T08:00:00.100+08:00|          2.0|
|1970-01-01T08:00:00.200+08:00|          2.0|
+-----------------------------+-------------+
Total line number = 2
It costs 0.023s
```

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java [hunk 1]
```diff
@@ -260,6 +262,9 @@ private PhysicalPlan doTransformation(Operator operator, int fetchSize)
       case TRACING:
         TracingOperator tracingOperator = (TracingOperator) operator;
         return new TracingPlan(tracingOperator.isTracingOn());
+      case SET_READONLY:
+        SetReadOnlyOperator setReadOnlyOperator = (SetReadOnlyOperator) operator;
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java:266]:
```suggestion
        SetSystemModeOperator setSystemModeOperator = (SetSystemModeOperator) operator;
```

Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/qp/strategy/PhysicalGenerator.java]: Fixed

**Comment by ijihang** [comment]
Can use this command in  distributed version ?

**Comment by HTHou** [comment]
> Can use this command in distributed version ?

I didn't try, but I think this new Plan needs to be adapted in distributed version just like other maintenance commands.

Are you interested in doing that adaptation？

**Comment by ijihang** [comment]
> I didn't try, but I think this new Plan needs to be adapted in distributed version just like other maintenance commands.
>
> Are you interested in doing that adaptation？

ok, I tried to adapt this command in distributed version

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**